types/row_proof.go

```diff
@@ -25,6 +25,9 @@ type RowProof struct {
 // the proof fails validation. If the proof passes validation, this function
 // attempts to verify the proof. It returns nil if the proof is valid.
 func (rp RowProof) Validate(root []byte) error {
+	if rp.EndRow < rp.StartRow {
+		return fmt.Errorf("end row %d cannot be smaller than start row %d", rp.EndRow, rp.StartRow)
+	}
 	// HACKHACK performing subtraction with unsigned integers is unsafe.
```

Comment:
[question] do we need this comment any more?

```suggestion
```

Reply:
Line 32 is only unsafe if `rp.EndRow-rp.StartRow+1` evaluates to less than 0 which shouldn't be possible because of the check here so I think the comment is safe to delete.

Alternatively we could convert the unsigned integers to signed integers before performing subtraction.

```go
	numberOfRows := int(rp.EndRow) - int(rp.StartRow) + 1
	if numberOfRows != len(rp.RowRoots) {
		return fmt.Errorf("the number of rows %d must equal the number of row roots %d", numberOfRows, len(rp.RowRoots))
	}
```